src/receivers/SocketModeReceiver.ts

```diff
@@ -187,10 +188,13 @@ export default class SocketModeReceiver implements Receiver {
 
         // Handle request for custom routes
         if (customRoutes.length && req.url) {
-          const match = this.routes[req.url] && this.routes[req.url][method] !== undefined;
+          // NOTE: the domain and scheme are irrelevant here.
+          // The URL object is only used to safely obtain the path to match
+          const { pathname: path } = new URL(req.url as string, 'http://localhost');
```

Comment:
We probably want to switch this hard-coded string out for the same `http://${req.headers.host}` that's featured in the `HTTPReceiver` implementation.

I believe that as a result, one of the tests will fail and the `fakeReq` will need to be augmented with the `headers` object (similar to what's seen in the test you've introduced).

Reply:
I actually changed it from that because it made it look like it had a purpose. As the comment on the other side explained

```
    // NOTE: the domain and scheme of the following URL object are not necessarily accurate. The URL object is only
    // meant to be used to parse the path and query
```

it doesn't actually serve any purpose beyond extracting the pathname, it's just there because URL _needs_ a URL and not a URI. 

I don't feel too strongly though, let me know if you still want it that way, I can fix it tomorrow.

Reply:
@moustacheful Thanks for elaborating here! 

You're right that, for the second argument in the `URL` constructor, choosing either `localhost` or `req.headers.host` does not matter in this scenario. That being said, we don't have a strong reason to use `localhost` (I know that's a common code example, though). Plus, we prefer consistency across the code base in general. 

Could you do the same with the code in `HTTPReceiver` tomorrow? Thanks again for taking the time to report and fix this issue!

Reply:
Hey yeah, no problem, I've updated `HTTPReceiver` to be consistent with `SocketReceiver`, and added a test case for this scenario on the former.